### DISCUSSION TITLE
Deploying Next.js on a VPS

### DISCUSSION THREAD

**Discussion by miguel9554** [vercel/next.js#15587] [Help] [answered]
Can anyone help me on how would a Next.js app be deployed to a generic VPS? Unfortunatley I can't deploy to Vercel, so I would have to go with the VPS, but I'm not finding much resources on the topic. I've gone through the official documentation, but the [Deployment](https://nextjs.org/docs/deployment) section only describes the deploy process with Vercel. 

Any help or link to documentation is appreciated, and also comments on wheter this is a good solution.

**Comment by jameswritescode** [comment, marked as answer]
This isn't a full breakdown, but the gist of it is still on that page: https://nextjs.org/docs/deployment#other-hosting-options

On your VPS you can build your app with `npm build` and then start it with `npm start`

**Reply by Lamndj** [reply]
running npm start will run the application but as soon as we close our ssh connection, it will stop. I guess we can it with forever package to make it run in background but again, can you run "npm start" with forever?

**Reply by farhankk360** [reply]
> running npm start will run the application but as soon as we close our ssh connection, it will stop. I guess we can it with forever package to make it run in background but again, can you run "npm start" with forever?

This can help: https://medium.com/codebase/how-to-keep-multiple-linux-terminals-running-in-background-screen-ccf2e53b0d22

**Comment by IRediTOTO** [comment]
VPS pretty hard to config, I did that before but took me some days to search google :D I deploy to a AWS lightsail.

**Reply by Lamndj** [reply]
I found the command after some research to run "npm start" script with forever : forever start -c "npm start" /path/to/app/dir/

And yes, VPS is pretty hard to set up. Cpanels are sweet.

**Comment by Rechdan** [comment]
For deploying to a VPS I strongly recommend using Dokku, its mostly all automatic and you just need to configure your Dokku App.

**Comment by M1ck0** [comment]
If you can't use `npm start` you can use `pm2` package. It will keep `npm` process running forever and it will also restart it if the process died for some reason

https://pm2.keymetrics.io/

**Comment by M1ck0** [comment]
For anyone that is looking for more native solution you can check [this](https://micko.dev/post/how-to-start-your-project-using-systemd-service)

**Reply by M1ck0** [reply]
This doesn't require any external packages, which means that no extra memory/cpu is taken from machine
